tiledb/sm/array_schema/dimension.cc

```diff
@@ -1379,62 +1379,13 @@ Status Dimension::set_tile_extent(const ByteVecValue& tile_extent) {
 }
 
 Status Dimension::set_null_tile_extent_to_range() {
```

Comment:
We can probably use the opportunity to remove some status variables here and throw.